src/file.cpp

```diff
@@ -946,21 +981,21 @@ namespace libtorrent
 	{
 #if defined TORRENT_WINDOWS && !defined TORRENT_MINGW
 #if TORRENT_USE_WSTRING
-		wchar_t cwd[TORRENT_MAX_PATH];
-		_wgetcwd(cwd, sizeof(cwd) / sizeof(wchar_t));
+		wchar_t* cwd { ::_wgetcwd(nullptr, 0) };
+		if (cwd == nullptr) return "";
+		std::string ret { convert_from_wstring(cwd) };
 #else
-		char cwd[TORRENT_MAX_PATH];
-		_getcwd(cwd, sizeof(cwd));
+		char* cwd { ::_getcwd(nullptr, 0) };
+		if (cwd == nullptr) return "";
+		std::string ret { convert_from_native(cwd) };
 #endif // TORRENT_USE_WSTRING
 #else
-		char cwd[TORRENT_MAX_PATH];
-		if (getcwd(cwd, sizeof(cwd)) == nullptr) return "/";
-#endif
-#if defined TORRENT_WINDOWS && !defined TORRENT_MINGW && TORRENT_USE_WSTRING
-		return convert_from_wstring(cwd);
-#else
-		return convert_from_native(cwd);
+		char* cwd { ::getcwd(nullptr, 0) };
+		if (cwd == nullptr) return "/";
+		std::string ret { convert_from_native(cwd) };
 #endif
+		std::free(cwd);
```

Comment:
ah, here it is. this is not exception safe. could you capture the ``cwd`` pointer in a RAII deleter? ``std::shared_ptr<char*> holder(cwd, &std::free);`` should work I believe